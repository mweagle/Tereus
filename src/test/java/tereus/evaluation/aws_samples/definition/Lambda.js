/* global CloudFormationTemplate,EC2 */
CloudFormationTemplate("EC2InstanceWithSecurityGroupSample")({
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "Validate the Lambda function",
  "Resources" :
  {
    "LambdaTest" : AWS.Lambda.Function(
     {
    	    "Code" : "./resources/simple",
    	    "S3Key" : "LambdaTest.zip",
    	    "Description" : "Simple AWS Lambda function",
    	    "Handler" : "index.handler"
     })
  }
});
