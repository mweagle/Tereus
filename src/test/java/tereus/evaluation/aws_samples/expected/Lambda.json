{
 "AWSTemplateFormatVersion": "2010-09-09",
 "Description": "Validate the Lambda function",
 "Parameters": {
  "BucketName": {
   "Description": "Name of an S3 bucket that will be used for Stack related resources",
   "Type": "String",
   "Default": "S3 Bucket Name in Stack Region",
   "MinLength": "1",
   "MaxLength": "255",
   "AllowedPattern": "[\\x20-\\x7E]*",
   "ConstraintDescription": "can contain only ASCII characters."
  }
 },
 "Resources": {
  "LambdaRole": {
   "Type": "AWS::IAM::Role",
   "Properties": {
    "AssumeRolePolicyDocument": {
     "Version": "2012-10-17",
     "Statement": [
      {
       "Effect": "Allow",
       "Action": [
        "sts:AssumeRole"
       ],
       "Principal": {
        "Service": [
         "lambda.amazonaws.com"
        ]
       }
      }
     ]
    },
    "Path": "/"
   }
  },
  "LambdaTest": {
   "Type": "AWS::Lambda::Function",
   "Properties": {
    "Runtime": "nodejs",
    "Code": {
     "S3Bucket": "testBucket",
     "S3Key": "LambdaTest.zip"
    },
    "Role": {
     "Fn::GetAtt": [
      "LambdaRole",
      "Arn"
     ]
    },
    "Description": "Simple AWS Lambda function",
    "Handler": "index.handler"
   }
  }
 }
}
